### PR TITLE
8290530: Bump minimum JDK version for JavaFX to JDK 17

### PR DESCRIPTION
**Pull request by kevinrushforth**
This PR bumps the minimum JDK version needed to build and run JavaFX to JDK 17, starting with JavaFX 20.

As discussed in [this thread](https://mail.openjdk.org/pipermail/openjfx-dev/2022-July/034874.html) on the `openjfx-dev` mailing list, this means that we can start using language features and APIs available from JDK 12 through 17. However, this is not an open invitation to globally refactor data classes into records, or change all multi-line Strings into text blocks. Rather we will be able to start using recent JDK capabilities in new code where it makes sense to do so.

The changes are as follows:

1. Added a new `jfx.jdk.target.version` property in `build.properties` to define the minimum JDK version, and set it to 17.
2. Bumped the minimum version of the boot JDK to 17 and the minimum version of gradle to 7.3.
3. Use the recent gradle feature to implement the minimum JDK version using `--release NN` instead of `-source NN -target NN` where possible (everywhere except when building the `javafx.swing` module and the system tests).
4. Update `CONTRIBUTING.md` to say that JDK 17 is the minimum.

Because this change has behavioral compatibility impact, it will need a CSR and a release note.

/reviewers 2
/csr

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires a CSR request to be approved
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issues
 * [JDK-8290530](https://bugs.openjdk.org/browse/JDK-8290530): Bump minimum JDK version for JavaFX to JDK 17
 * [JDK-8291052](https://bugs.openjdk.org/browse/JDK-8291052): Bump minimum JDK version for JavaFX to JDK 17 (**CSR**)

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)
 * [Johan Vos](https://openjdk.org/census#jvos) (@johanvos - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx pull/850/head:pull/850` \
`$ git checkout pull/850`

Update a local copy of the PR: \
`$ git checkout pull/850` \
`$ git pull https://git.openjdk.org/jfx pull/850/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 850`

View PR using the GUI difftool: \
`$ git pr show -t 850`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/850.diff">https://git.openjdk.org/jfx/pull/850.diff</a>

</details>

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -627,8 +630,8 @@ try {
             defineProperty("jdkBuildNumber", jdkVersionInfo[1])
 
             // Define global properties based on the version of Java
-            def status = compareJdkVersion(jdkVersion, "12")
-            ext.jdk12OrLater = (status >= 0)
+//            def status = compareJdkVersion(jdkVersion, "18")
+//            ext.jdk18OrLater = (status >= 0)
```
Comment (prrace) [inline build.gradle:634]: Why do we have this if it is commented out ?

Comment (kevinrushforth) [reply build.gradle:634]: It's meant to be an example of how to do a conditional based on the JDK version. I could have just removed it, but I'll add a comment to make that clear.

**Comment by kevinrushforth** [comment]
I see that the GHA test run didn't start on macOS, likely due to the deprecation of the macOS 10.15 platform. I'm merging in the latest master to pick up the recently-integrated fix to use macOS 11 instead.

File: build.gradle [hunk 2]
```diff
@@ -2626,6 +2638,12 @@ project(":controls") {
 
 project(":swing") {
 
+    // We need to skip setting compiler.options.release for this module,
+    // since javafx.swing requires jdk.unsupoprted.desktop, which is
```
Comment (prrace) [inline build.gradle:2642]: jdk.unsupoprted.desktop

typo !

**Review by prrace — APPROVED** [review]

File: build.gradle (hunk 2 above)
Comment (kevinrushforth) [reply build.gradle]: Fixed. I had the same typo in the CSR (copy/paste) so I fixed it there, too.

Thanks.

**Review by prrace — APPROVED** [review]

**Review by johanvos — APPROVED** [review]

**Pull request closed without merging**